tests/sql/test_catalog.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+
+import daft
+from daft.sql import SQLCatalog
+
+
+@pytest.mark.parametrize(
+    "table_name",
+    ["a", "a_b", '"a.b"', '"a.b.c"', "a_", "a_1", '"a-b"', '"a."'],
+)
+def test_sql_catalog_table_names(table_name):
+    df1 = daft.from_pydict({"idx": [1, 2], "val": [10, 20]})
+    catalog = {table_name: df1}
+    try:
+        actual = daft.sql(f"select * from {table_name}", catalog=SQLCatalog(catalog)).to_pydict()
```

Comment:
I believe "a, a_b, a_, a_1" are all valid table names. To be honest, these should all succeed when we delimit the identifier. The catalog should not be doing the validation and should accept names as is. We can sync today on this behavior because the handling of identifiers is a bit more nuanced than this.

Reply:
Not sure i understand. this is the passing test cases. All of these are valid table names. 
Good: `["a", "a_b", '"a.b"', '"a.b.c"', "a_", "a_1", '"a-b"', '"a."'],`
Bad:  `["1", "a.", "a b", "a-b"]`

as those are not proper identifiers in most systems unless properly quoted. 

ex: 
```sql
create table a; -- valid
create table a_b; -- valid
create table "a.b"; -- valid
create table "a.b.c"; -- valid

-- ....

create table 1;  -- invalid as 1 is a scalar value
create table a.; -- invalid 
create table a b; -- invalid
create table a-b; -- invalid (arguably, this _could_ be valid, but we don't support it)
```

Reply:
For sure, I was not able to see the additional quotes in the parameters so thought valid and invalid were being mixed here.